kubernetes/nodecollection.go

```diff
@@ -110,7 +110,8 @@ func downloadNodeData(prefix string,
 
 	for _, n := range nodes {
 		if n.Spec.ProviderID == "" {
-			failedNodeList[n.Name] = errors.New("Provider ID for node does not exist. " +
+			log.Warnf("Node ProviderID is not set, may cause inconsistent gathering of metrics data.")
+			failedNodeList[n.Name] = errors.New("provider ID for node does not exist. " +
```

Comment:
Do we miss these failedNodeList errors in the logs? The new log statement seems similar, and I'm just wondering if we'll be double logging the same issue here when Nodes fail (because I think parent calls to this method log the error). Is there some more nuance to this to consider when we'd be trying to triage a customer issue where the additional log statement tells us more?

Reply:
The second error will output a list of all of the nodes that have errors with the nodes.  I'll update this message to make it clearer because this usually happens in cases where the node is in a self managed environment (i.e. not on AKS, EKS, or GKE)

Reply:
Sounds good, adding a little color to the log.Warn message will make it more useful to read in addition to the list of failed nodes with the more generic "provider id doesn't exist, may cause more errors"